extensions/file-manager/src/utils.tsx

```diff
@@ -38,8 +37,59 @@ export type PreferencesType = {
   showFilePermissions: boolean;
   showFileSize: boolean;
   startDirectory: string;
+  showDeleteOption: boolean;
 };
 
+export async function deleteFile(filePath: string, fileName: string, refresh: () => void) {
+  const options: Alert.Options = {
+    title: "Delete File?",
+    message: `Are you sure you want to delete ${fileName}?`,
+    icon: Icon.Eraser,
+    primaryAction: {
+      title: "Delete",
+      style: Alert.ActionStyle.Destructive,
+      onAction: async () => {
+        await runShellScript(`rm -rf "${filePath}"`);
+        refresh();
+        showToast(Toast.Style.Success, "File Deleted", `${fileName}`);
+      },
+    },
+  };
+
+  await confirmAlert(options);
+
+  return;
+}
```

Comment:
Given the delete actions are permanent, it'd be better to let users know what they're doing is irreversible.

```suggestion
export async function deleteFile(filePath: string, fileName: string, refresh: () => void) {
  const options: Alert.Options = {
    title: "Permanently Delete File?",
    message: `Are you sure you want to delete ${fileName}? This action cannot be undone.`,
    icon: Icon.Eraser,
    primaryAction: {
      title: "Delete",
      style: Alert.ActionStyle.Destructive,
      onAction: async () => {
        await runShellScript(`rm -rf "${filePath}"`);
        refresh();
        showToast(Toast.Style.Success, "File Deleted", `${fileName}`);
      },
    },
  };

  await confirmAlert(options);

  return;
}
```